Board.cpp

```diff
@@ -153,20 +155,19 @@ bool Board::checkBlockFieldColision() {
 }
 
 void Board::generateNewBlock() {
-	if (bomb) {
-		delete bomb;
-		bomb = nullptr;
-	}
-	else if (block)
+	if (block)
 		delete block;
-	if (steps % 20 == 0) {
-		bomb = new Bomb(Point(boardOffset.getX() + WIDTH / 2 - 2, 0));
-		block = bomb;

```

Comment:
its bad practice to have a different pointer for a erived class. we should use dynamic casting when needed a specific bomb stuff